### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by AnielaMW**
The user profile page has been created and completed. 
I'm trying to get better at Git. Please let me know were I am still going wrong.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,3 +1,22 @@
 class UsersController < ApplicationController
-  
+  before_action :set_user, only: [:show]
+
+  def show
+    @dances = list_dances.sort_by { |d| d.title }
+    @dance_comments = list_dcomments.sort_by { |d| d.created_at }.reverse
+  end
```
Comment (itiswicked) [inline app/controllers/users_controller.rb]: You may be able to replace the ruby sorting with ORDER BY at the database level:

```
list_dances.order(:title)
list_dcomments.order(created_at: :asc) # or :desc
```

https://apidock.com/rails/ActiveRecord/QueryMethods/order

File: app/models/user.rb [hunk 2]
```diff
@@ -9,4 +9,12 @@ class User < ApplicationRecord
...
   validates :last_name, presence: true
+
+  def dances?
+    dances != []
+  end
+
+  def dance_comments?
+    dance_comments != []
+  end
 end
```
Comment (itiswicked) [inline app/models/user.rb:20]: nice abstraction here

File: spec/features/dance_comments/4_user_updates_spec.rb [hunk 3]
```diff
@@ -67,6 +67,8 @@
     click_button "Update"
 
     expect(page).to have_content("Comment can't be blank")
+
+    # CANNOT SEEM TO TEST FOR TYPE CAN'T BE BLANK.
   end
```
Comment (itiswicked) [inline spec/features/dance_comments/4_user_updates_spec.rb:72]: Hmm, you might have to use `unselect` here. Requires knowing which option is selected. 
http://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FNode%2FActions%3Aunselect

File: spec/features/dance_comments/3_user_creates_spec.rb [hunk 4]
```diff
@@ -41,9 +41,8 @@
     click_link "Create Comment"
     click_button "Create Dance comment"
 
-    expect(page).to have_content(
-      "Comment can't be blank"
-    )
+    expect(page).to have_content("Comment can't be blank")
+
     # CANNOT SEEM TO TEST FOR TYPE CAN'T BE BLANK.
```
Comment (itiswicked) [inline spec/features/dance_comments/3_user_creates_spec.rb:46]: Hmm, your associated code looks good. Perhaps the type error is being produced but is not being matched by the `expect` you had written? I would def reach out to chat support to troubleshoot this together.

**Comment by itiswicked** [comment]
Good stuff!!!

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (AnielaMW) [reply app/controllers/users_controller.rb]: How would that work if it's pulling the dances or comments through the models?

**Pull request merged**